src/homework6/classes.py

```diff
@@ -0,0 +1,166 @@
+"""Create simulator from real life. This can be booking room in hotel,
+visit to casino, visit to bar. Create 3-4 objects, that can describe situation
+Objects should contain attributes and methods to simulate some use cases.
+Completed program should print object states, it actions (methods)
+and objects interaction."""
+
+
+class Vehicle(object):
+    """Describes vehicle with model, key and size.
+
+    Key is unique for each vehicle.
+    Size shows how much space vehicle will occupy in parking lot.
+    """
+    def __init__(self, model, size):
+        self.model = model
+        self.size = size
+        self.key = id(self)
+
+    def __str__(self):
+        return self.model
+
+
+class Driver(object):
+    """Describes person who owns vehicle and key for it.
+
+    He can buy, sell, park and pick up from parking lot vehicle.
+    """
+    def __init__(self, name, vehicle=None):
+        self.name = name
+        self.vehicle = vehicle
+        if vehicle:
+            self.key = vehicle.key
+        else:
+            self.key = None
+
+    def buy_vehicle(self, vehicle):
+        """Buy vehicle.
+
+        Driver won't be able to buy a new vehicle until he sells his.
+        """
+        if self.key:
+            print(self.name, "already has vehicle.".format(self.name))
+        else:
+            self.vehicle = vehicle
+            self.key = vehicle.key
+            print("{} bought {}".format(self.name, vehicle))
+
+    def sell_vehicle(self):
+        """Sell vehicle.
+
+        Driver won't be able to sell his vehicle
+        until it is in the parking lot.
+        """
+        if self.vehicle:
+            self.vehicle = None
+            self.key = None
+            print("{} sold {}".format(self.name, self.vehicle))
+        elif self.key:
+            print(self.name, "can't sell vehicle, it is in the parking lot.")
+        else:
+            print(self.name, "has no vehicle.")
+
+    def park_vehicle(self, parking):
+        """Park vehicle int the parking lot.
+
+        Driver can't park vehicle if parking has not enough space.
+        """
+        if not self.key:
+            print(self.name, "has no vehicle.")
+        elif not self.vehicle:
```

Comment:
it seems to be strange name - I expect his vehicle here but sometimes it's none.